.github/workflows/build.yml

```
@@ -4,233 +4,187 @@ on:
+      run: |
+        # Decrypt the certificate
+        openssl enc -aes-256-cbc -d -md sha512 \
+          -k "${MACOS_CODESIGN_OPENSSL_PASSWORD}" \
+          -in /Users/runner/work/mixxx/mixxx/cmake/macos_developer_id_codesign_certificate.p12.enc \
+          -out ~/certificate.p12
+
+        # Create a temporary keychain for the certificate and import it.
+        security create-keychain -p mixxx Mixxx.keychain
+        security unlock-keychain -p mixxx Mixxx.keychain
+        security import ~/certificate.p12 -k Mixxx.keychain \
+          -P "${MACOS_CODESIGN_CERTIFICATE_PASSWORD}" -T /usr/bin/codesign
+        security set-key-partition-list -S "apple-tool:,apple:" -k mixxx Mixxx.keychain
+        # Add keychain to search list
+        security list-keychains -s Mixxx.keychain
+
+        CMAKE_ARGS="-DAPPLE_CODESIGN_IDENTITY=${APPLE_CODESIGN_IDENTITY}"
+        echo "::set-output name=cmake_args::${CMAKE_ARGS}"
+
+    - name: "Set up build environment"
+      id: buildenv
+      run: ${{ matrix.buildenv_script }} setup --ghactions
+      env:
+        BUILDENV_BASEPATH: ${{ github.workspace }}/buildenv
+
+    - name: "Set up ccache"
+      run: ccache -M 5G
+      env:
+        PATH: ${{ steps.buildenv.outputs.path }}
+
+    - name: "Set up ccache cache"
+      uses: actions/cache@v2
+      with:
+        path: ${{ matrix.ccache_path }}
         key: ${{ matrix.os }}-ccache-${{ github.head_ref }}-${{ github.run_number }}
         restore-keys: |
           ${{ matrix.os }}-ccache-${{ github.head_ref }}
           ${{ matrix.os }}-ccache
 
-    - name: Preconfigure Ubuntu 18.04
-      if: matrix.os == 'ubuntu-18.04'
-      run: echo "CMAKE_FLAGS_UBUNTU_EXTRA=-DFAAD=ON" >> $GITHUB_ENV
-
-    - name: Preconfigure Ubuntu 20.04
-      if: matrix.os == 'ubuntu-20.04'
-      run: echo "CMAKE_FLAGS_UBUNTU_EXTRA=-DFFMPEG=ON" >> $GITHUB_ENV
+    - name: "Create build directory"
+      run: mkdir cmake_build
 
-    - name: Configure
-      run: cmake $CMAKE_FLAGS_COMMON $CMAKE_FLAGS_UBUNTU $CMAKE_FLAGS_UBUNTU_EXTRA -S . -B cmake_build
+    - name: "Configure"
+      run: |
+        cmake \
+          -DCMAKE_BUILD_TYPE=Release \
+          -DBATTERY=ON \
+          -DBROADCAST=ON \
+          -DBULK=ON \
+          -DDEBUG_ASSERTIONS_FATAL=ON \
+          -DHID=ON \
+          -DLILV=ON \
+          -DOPUS=ON \
+          -DQTKEYCHAIN=ON \
+          -DVINYLCONTROL=ON \
+          ${CMAKE_FLAGS_EXTRA} \
+          ..
+      working-directory: cmake_build
       env:
-         CMAKE_FLAGS_UBUNTU: -DKEYFINDER=ON -DLOCALECOMPARE=ON -DMAD=ON -DMODPLUG=ON -DWAVPACK=ON
+         PATH: ${{ steps.buildenv.outputs.path }}
+         CMAKE_FLAGS_EXTRA: ${{ matrix.cmake_args }} ${{ steps.apple_codesign.outputs.cmake_args }} -DCMAKE_PREFIX_PATH=${{ steps.buildenv.outputs.cmake_prefix_path }} -DQt5_DIR=${{ steps.buildenv.outputs.qt_path }}
```

```suggestion
         CMAKE_FLAGS_EXTRA: ${{ matrix.cmake_args }} ${{ steps.apple_codesign.outputs.CMAKE_ARGS }} -DCMAKE_PREFIX_PATH=${{ steps.buildenv.outputs.cmake_prefix_path }} -DQt5_DIR=${{ steps.buildenv.outputs.qt_path }}
```

Are the environment variable names case sensitive?

---

I think so. Anyway, these are not environment variables but output parameters: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#setting-an-output-parameter